packages/expect/src/matchers.ts

```diff
@@ -62,32 +70,57 @@ const matchers: MatchersObject = {
       ? () =>
           matcherHint(matcherName, undefined, undefined, options) +
           '\n\n' +
-          `Expected: ${printExpected(expected)}\n` +
-          `Received: ${printReceived(received)}`
+          `Expected: not ${printExpected(expected)}`
       : () => {
           const receivedType = getType(received);
           const expectedType = getType(expected);
-          const suggestToEqual =
-            receivedType === expectedType &&
-            (receivedType === 'object' || expectedType === 'array') &&
-            equals(received, expected, [iterableEquality]);
-          const oneline = isOneline(expected, received);
-          const diffString = diff(expected, received, {expand: this.expand});
+
+          let deepEqualityName = null;
+          if (expectedType !== 'map' && expectedType !== 'set') {
+            // If deep equality passes when referential identity fails,
+            // but exclude map and set until review of their equality logic.
+            if (equals(received, expected, toStrictEqualTesters, true)) {
+              deepEqualityName = 'toStrictEqual';
+            } else if (equals(received, expected, [iterableEquality])) {
+              deepEqualityName = 'toEqual';
+            }
+          }
+
+          const hasConciseReport =
+            receivedType !== expectedType ||
+            (isPrimitive(expected) &&
+              (expectedType !== 'string' || isOneline(expected, received))) ||
+            expectedType === 'date' ||
+            expectedType === 'function' ||
+            expectedType === 'regexp' ||
+            (received instanceof Error && expected instanceof Error);
+          const hasDifference = stringify(expected) !== stringify(received);
+          const difference =
+            !hasConciseReport && hasDifference
+              ? diff(expected, received, {expand: this.expand}) // string | null
+              : null;
 
           return (
             matcherHint(matcherName, undefined, undefined, options) +
             '\n\n' +
-            `Expected: ${printExpected(expected)}\n` +
-            `Received: ${printReceived(received)}` +
-            (diffString && !oneline ? `\n\nDifference:\n\n${diffString}` : '') +
-            (suggestToEqual ? ` ${SUGGEST_TO_EQUAL}` : '')
+            (deepEqualityName !== null
+              ? DIM_COLOR(
+                  `If it should pass with deep equality, replace toBe with ${deepEqualityName}`,
```

Comment:
How about using `${matcherName}` instead of `toBe` and wrapping matcher names in quotes so they're easily distinguishable from the rest of the text?
```suggestion
                  `If it should pass with deep equality, replace "${matcherName}" with "${deepEqualityName}"`,
```